### PR TITLE
Refactor docker image building scripts.

### PR DESCRIPTION
**Pull request by arkid15r**
## Issue Number

#3129 

## Purpose/Implementation Notes
Update scripts for better docker cache leveraging, improve readability:
  - Add remote builder support option.
  - Add existing image based cache.
  - Set `amd64` as a default architecture.
  - Add optional `arm64` build support for `update_my_docker_images.sh`.
  - Consolidate api/common/workers images build logic.

## Methods
NA

## Types of changes

What types of changes does your code introduce?

NA

## Functional tests
NA

## Checklist

- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Screenshots
NA

### CODE REVIEW

File: scripts/update_my_docker_images.sh [hunk 1]
```diff
@@ -13,39 +10,48 @@ script_directory="$(perl -e 'use File::Basename;
...
     cat << EOF
 There are two required arguments for this script:
--d specifies the Dockerhub repo you would like to deploy to.
+-d specifies the Dockerhub repository you would like to deploy to.
 -v specifies the version you would like to build. This version will passed into
     the Docker image as the environment variable SYSTEM_VERSION.
     It also will be used as the tag for the Docker images built.
 
-There is also one optional argument:
+There is also optional arguments:
```
Comment (davidsmejia) [inline scripts/update_my_docker_images.sh:29]:
```suggestion
There are also optional arguments:
```

File: scripts/update_my_docker_images.sh [hunk 2]
```diff
@@ -13,39 +10,48 @@ script_directory="$(perl -e 'use File::Basename;
...
 -a also build the affymetrix image
     (we normally don't because it is so intense to build)
+-b sets a remote Docker builder for amd64 image builds.
```
Comment (davidsmejia) [inline scripts/update_my_docker_images.sh:32]: We should simplify this, default to `adm64` and have optarg `-x` to target `arm64`. Though I also think we probably shouldn't enable this until all images can be built on `arm64`. Would like to hear your thoughts on this.

**Review by davidsmejia — CHANGES_REQUESTED** [review]
Default to amd64 and then have an optional flag to target arm

File: scripts/update_my_docker_images.sh (hunk 2 above)
Comment (davidsmejia) [reply scripts/update_my_docker_images.sh:32]: Actually I am unsure what this option does functionally.

**Comment by arkid15r** [comment]
Suppressed by https://github.com/AlexsLemonade/refinebio/pull/3285

**Pull request closed without merging**